use-cases/xtclim/src/trainer.py

```diff
@@ -14,56 +14,114 @@
 tracker.start()
 """
 
+from typing import Literal, Optional
 import torch
 import torch.optim as optim
 import numpy as np
 import pandas as pd
+import sys
+import os
+
+from itwinai.components import monitor_exec
+from itwinai.torch.trainer import TorchTrainer
+from itwinai.torch.distributed import (
+    distributed_resources_available,
+    TorchDistributedStrategy,
+    TorchDDPStrategy,
+    HorovodStrategy,
+    DeepSpeedStrategy,
+    NonDistributedStrategy
+)
+from itwinai.loggers import Logger
+from itwinai.torch.config import TrainingConfiguration
 
-from itwinai.components import Trainer, monitor_exec
 import model
 from torch.utils.data import DataLoader
 from torchvision.utils import make_grid
 from engine import train, validate
 from utils import save_reconstructed_images, save_loss_plot, save_ex
-from initialization import device, beta, criterion
+from initialization import beta, criterion, pixel_wise_criterion
 
-class TorchTrainer(Trainer):
+class XTClimTrainer(TorchTrainer):
     def __init__(
             self,
             epochs: int,
             batch_size: int,
-            lr: float
+            lr: float,
```

Comment:
`lr` and `batch_size` should be embedded in `config` (needs to be added to the constructor). The training config contains all the hyperparams and user-defined params

Reply:
Done

Reply:
this seems to be unresolved yet. Also, try as much as possible to reuse the existing field names from the training configuration: https://itwinai.readthedocs.io/latest/python-api/itwinai.torch.modules.html#itwinai.torch.config.TrainingConfiguration
Example, in this case `lr` would become `optim_lr` .

It is also possible to create a custom configuration for the use case to include use case hyperparams, as we did for Virgo: https://github.com/interTwin-eu/itwinai/blob/af68d9022e7d96312d84847834930abc786a98c5/use-cases/virgo/trainer.py#L36